sphinx/ext/autodoc/__init__.py

```diff
@@ -1473,12 +1468,8 @@ def format_args(self, **kwargs: Any) -> str:
             args = args.replace('\\', '\\\\')
         return args
 
-    def add_directive_header(self, sig: str) -> None:
-        meth = self.parent.__dict__.get(self.objpath[-1])
-        if inspect.is_singledispatch_method(meth):
```

Comment:
I don't fully understand what the difference was between:

* `meth`
* `self.object`
* `obj`

Based on the tests still passing, it seems that `meth` and `self.object` were the same?